plugins/performance-lab/includes/site-health/audit-enqueued-assets/hooks.php

```diff
@@ -13,107 +13,121 @@
 // @codeCoverageIgnoreEnd
 
 /**
- * Audit enqueued and printed scripts in is_front_page(). Ignore /wp-includes scripts.
+ * Audit blocking assets on the front page.
  *
- * It will save information in a transient for 12 hours.
- *
- * @since 1.0.0
- *
- * @global WP_Scripts $wp_scripts
+ * @since n.e.x.t
  */
-function perflab_aea_audit_enqueued_scripts(): void {
-	if ( ! is_admin() && is_front_page() && current_user_can( 'view_site_health_checks' ) && false === get_transient( 'aea_enqueued_front_page_scripts' ) ) {
-		global $wp_scripts;
-		$enqueued_scripts = array();
+function perflab_aea_audit_blocking_assets(): void {
+	if (
+		! is_admin() ||
+		! current_user_can( 'view_site_health_checks' ) ||
+		false !== get_transient( 'aea_blocking_assets' )
+	) {
+		return;
+	}
 
-		foreach ( $wp_scripts->done as $handle ) {
-			$script = $wp_scripts->registered[ $handle ];
+	$response = wp_remote_get(
+		home_url( '/' ),
+		array(
+			'timeout' => 10,
+			'headers' => perflab_aea_copy_basic_auth_headers(
+				array(
+					'Accept'        => 'text/html',
+					'Cache-Control' => 'no-cache',
+				)
+			),
+		)
+	);
 
-			if ( ! $script->src || false !== strpos( $script->src, 'wp-includes' ) ) {
-				continue;
-			}
+	if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+		return;
+	}
 
-			// Add any extra data (inlined) that was passed with the script.
-			$inline_size = 0;
-			if (
-				isset( $script->extra['after'] ) &&
-				is_array( $script->extra['after'] )
-			) {
-				foreach ( $script->extra['after'] as $extra ) {
-					$inline_size += ( is_string( $extra ) ) ? mb_strlen( $extra, '8bit' ) : 0;
-				}
-			}
+	$html = wp_remote_retrieve_body( $response );
+	if ( '' === $html ) {
+		return;
```

Comment:
Ditto. If the HTML response is empty, then this here would indicate a problem with the test that could be exposed.